lib/git_diff_parser/patch.rb

```diff
@@ -93,6 +93,8 @@ def removed_lines
           )
           lines << line
           line_number += 1
+        when NOT_REMOVED_LINE
```

Comment:
I could be wrong, but I think we should only be incrementing `line_number` on unchanged lines, i.e. `when /^\s/`.

Here is a fix and a failing test: https://github.com/cjoudrey/ruby-git_diff_parser/commit/a37ad989b4bfcc419b031a3a20ccbd5f0a6f94ed#diff-5167fa12a2c6d40aaa8884e40efa7ac4

**Explanation:**

Original file:

```graphql
type Query {
  a: String
  b: String
  c: String
}
```

Patch:

```diff
@@ -1,5 +1,5 @@
 type Query {
+  d: String
   a: String
   b: String
-  c: String
 }
```

The number of the line being removed here is number `4` and patch position `5`.

With this current pull request, the number is `5` and patch position is `5`.